(#3879) Add aria-labelledby to components

dev-rana-publicis opened the pull request:
Closes #3879
ODE: https://ncigovcdode579.prod.acquia-sites.com/

bennettcc commented:
Andy approved this PR on 6/9 - need one more product reviewer then the checkmark should be added

sarinapadilla reviewed (approved):
Looks good to product!

dev-rana-publicis commented:
Current Changes:

- NCIDS Hero
  - changed from `<section>` to `<div>`
  - no longer has `aria-labelledby` (was on the `section` prior to the `div` change) and no longer has `id` on the Hero's heading
- NCIDS Call to Action Strip (Standalone)
  - changed from `<section>` to `<div>`
  - does not have `aria-labelledby`

- Call to Action Strip (Hero variant & standalone)
  - Bryan noticed (while looking at ACSF on his phone) that the link button text was not the correct color
  - This PR fixes the link button text color

bryanpizzillo commented on docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-2-guide-card-row.html.twig:
```diff
@@ -1,5 +1,8 @@
+  {% set id = paragraph.id() %}
+  {% set entity_type = paragraph.entityTypeID %}
+  {% set titleID = "#{entity_type}-#{id}" %}
+  <section class="usa-section usa-section--light cgdp-guide-card-row" aria-labelledby="{{ titleID }}" data-eddl-landing-row>
```
This seems wrong. Nothing has this id.

bryanpizzillo reviewed (changes requested):
Can you please look into the thing below?
